### PR TITLE
network policy workload

### PR DESCRIPTION
**Pull request by venkataanil**
Network policy testing touches 3 key areas
1. Template to include all network policy configuration options
2. Latency measurement through connection testing
3. Flow tracking through Convergence tracker

This PR is adding templates which proposed unified approach to add namespace and pod selectors, CIDRs, single ports and port ranges. CIDRs and egress will be added in next patch.

Legacy network policy templates will be retired and removed once this new approach gets stabilized.

This PR depends on following kube-burner and images PRs for connection testing
https://github.com/cloud-bulldozer/images/pull/13
https://github.com/cloud-bulldozer/images/pull/12
https://github.com/kube-burner/kube-burner/pull/679

A follow up PR will be submitted for adding Converagnce tracker
 tool support.

More details are added in README.md

### CODE REVIEW

**Comment by venkataanil** [comment]
lint errors are related kube-burner dependeny. So ignore them for now.

File: network-policy.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	Spec: routev1.RouteSpec{
+		Port: &routev1.RoutePort{TargetPort: intstr.FromString("http")},
+		To: routev1.RouteTargetReference{
+			Name: networkPolicyProxy,
+		},
+	},
+}
+
+// create proxy pod with route
+func deployAssets(uuid string, clientSet kubernetes.Interface, restConfig *rest.Config) error {
```
Comment (vishnuchalla) [inline network-policy.go:106]: Can we move all these assets to the config as well? If we intent to run them at start only once we can do that by either of the options below
* [Add them to start of config](https://github.com/kube-burner/kube-burner-ocp/blob/main/cmd/config/networkpolicy-matchlabels/networkpolicy-matchlabels.yml#L26-L34)

   OR

* [By using run once option we have](https://kube-burner.github.io/kube-burner/latest/reference/configuration/?h=runonce#runonce)

Comment (venkataanil) [reply network-policy.go:106]: Thanks Vishnu. Initially I was doing the same 
https://github.com/venkataanil/files/blob/master/kube-burrner-netpol/network-policy.yml#L11 
However, the route will be used inside core kube-burner measurement code, so earlier I was manually adding it in the core kube-burner measurement code
https://github.com/venkataanil/files/blob/master/kube-burrner-netpol/netpol_latency.go#L65

Now kube-burner-ocp creates it and passes it as environment variable https://github.com/kube-burner/kube-burner-ocp/pull/117/files#diff-43d885d93cbe44660401e7017fb54c6cfe09507946b9374ad7d7b0a0485e5143R172 
This env variable is consumed in the template
https://github.com/kube-burner/kube-burner-ocp/pull/117/files#diff-22fffda545f823bb596360df5ac5fffdbd6b233b9b662a67ddd506d48162d677R11
Finally core kube-burner measurement code reads from the config file https://github.com/kube-burner/kube-burner/pull/679/files#diff-a6045e2aa0d1b86a2c266acf164b4102c828a73863f6212d7eff79a34be53e04R496 

I was following similar approach of service_latency.go https://github.com/kube-burner/kube-burner/blob/main/pkg/measurements/service_latency.go#L73 
However I couldn't directly define deploy_assets() which creates this proxy route in 
 pkg/measurements/netpol_latency.go (https://github.com/kube-burner/kube-burner/pull/679/files#diff-a6045e2aa0d1b86a2c266acf164b4102c828a73863f6212d7eff79a34be53e04) as "route" resource belongs to ocp and not k8s. So I needed to move it to kube-burner-ocp.

I can move it to template if there is a way measurement code pkg/measurements/netplo_latency.go get this route information.

Comment (vishnuchalla) [reply network-policy.go:106]: Thanks for the detailed explanation. My concern was to just move these assets creation to the top of the workload config file in a desired order but not to kube-burner core repo.

Comment (venkataanil) [reply network-policy.go]: This is moved now.

**Review by mohit-sheth — APPROVED** [review]
thanks @venkataanil, lgtm

**Review by vishnuchalla — APPROVED** [review]
lgtm

**Comment by rsevilla87** [comment]
> @vishnuchalla [kube-burner/kube-burner#679](https://github.com/kube-burner/kube-burner/pull/679) got merged. Do we need for a new kube-burner release to get this merged?

yes, https://github.com/kube-burner/kube-burner/releases/tag/v1.11.1

**Pull request merged**